src/SprykerSdk/AsyncApi/Code/Builder/AsyncApiCodeBuilder.php

```diff
@@ -178,79 +180,195 @@ protected function createTransferForMessage(
-            $asyncApiResponseTransfer->addMessage($this->messageBuilder->buildMessage(AsyncApiInfo::addedPropertyWithTypeTo($propertyName, $type, $asyncApiMessageName, $moduleName)));
-        }
+        $asyncApiResponseTransfer->addMessage($this->messageBuilder->buildMessage(AsyncApiInfo::addedPropertyWithTypeTo('messageAttributes', 'MessageAttributesTransfer', $asyncApiMessageName, $moduleName)));
 
-        $transferBuildCommandLine = [
+        $commandLines[] = [
             $this->config->getSprykRunExecutablePath() . '/vendor/bin/spryk-run',
-            'AddSharedTransferProperty',
+            'AddSharedTransferDefinition',
             '--mode', $this->sprykMode,
             '--organization', $projectNamespace,
             '--module', $moduleName,
-            '--name', $asyncApiMessageName,
-            '--propertyName', implode(',', $transferPropertiesToAdd),
+            '--name', 'MessageAttributes',
             '-n',
             '-v',
         ];
 
-        $commandLines[] = $transferBuildCommandLine;
+        $asyncApiResponseTransfer->addMessage($this->messageBuilder->buildMessage(AsyncApiInfo::addedTransferDefinitionTo('MessageAttributeTransfer', $moduleName)));
 
-        // Add messageAttributes to the Transfer
-        $commandLines[] = [
+        $this->runCommandLines($commandLines);
+
+        return $asyncApiResponseTransfer;
+    }
+
+    /**
+     * In previous versions we used the operationId as attribute to get where to render the code to we have to support it for BC reasons.
+     *
+     * We introduced a new `x-spryker` extension for the schema file:
+     *
+     * ```
+     * x-spryker:
+     *      module: ModuleName
+     * ```
+     *
+     * @param \SprykerSdk\AsyncApi\AsyncApi\Message\AsyncApiMessageInterface $asyncApiMessage
+     *
+     * @return string
+     */
+    protected function getModuleNameForMessage(AsyncApiMessageInterface $asyncApiMessage): string
+    {
+        if ($asyncApiMessage->getAttribute('x-spryker')) {
+            /** @var \SprykerSdk\AsyncApi\AsyncApi\Message\Attributes\AsyncApiMessageAttributeCollectionInterface $xSprykerAttributeCollection */
+            $xSprykerAttributeCollection = $asyncApiMessage->getAttribute('x-spryker');
+
+            /** @var \SprykerSdk\AsyncApi\AsyncApi\Message\Attributes\AsyncApiMessageAttributeInterface|null $moduleNameAttribute */
+            $moduleNameAttribute = $xSprykerAttributeCollection->getAttribute('module');
+
+            if ($moduleNameAttribute) {
+                /** @var string $moduleName */
+                $moduleName = $moduleNameAttribute->getValue();
+
+                return $moduleName;
+            }
+        }
+
+        // BC-Fallback to operationId
+        /** @var \SprykerSdk\AsyncApi\AsyncApi\Message\Attributes\AsyncApiMessageAttributeInterface $operationIdAttribute */
+        $operationIdAttribute = $asyncApiMessage->getAttribute('operationId');
```

Comment:
are you sure that we need to support old format? it can conflicts if people will use operationId according to the specification, e.g. `emitUserSignUpEvent`

I would release version 0.3 without this.